### ISSUE TITLE
Ensure we're testing the fastest possible TorchScript gradient

### ISSUE THREAD

**Issue by awf** [microsoft/knossos-ksc#614]
TorchScript does have "real" s2s AD -- see call to differentiate at (https://github.com/pytorch/pytorch/blob/1d6bd157902d4b1347a5d03122d02b407658e263/torch/csrc/jit/runtime/graph_executor.cpp#L685)

We should check we're calling this, e.g. through _jit_differentiate, or by ensuring requires_grad=True before jitting.

We should get GRAPH_DEBUG on, and emit graphs as Python code.  If those are compilable, then sending them to npcomp may the most realistic (and testing) baseline for us.
